cli/parser/options/options.go

```diff
@@ -289,19 +287,145 @@ type Option interface {
 	BoolLike() BoolLike
 }
 
+// The base type for all option types. Handles a lot of formatting and option
+// representation transformations.
+type optionBase struct {
+	*Definition
+
+	// The name this flag will use when formatted.
+	Form flag_form.Form
+}
+
+// Returns a new option base derived from the option string (without the leading
+// dashes) and the provided Definition.
+func NewOptionBase(opt string, d *Definition) (*optionBase, error) {
+	// validate optName
+	var form flag_form.Form
+	switch opt {
+	case d.name:
+		form = flag_form.Standard
+	case d.shortName:
+		form = flag_form.Short
+	default:
+		if d.hasNegative {
+			if n, cut := strings.CutPrefix(opt, "no"); cut {
+				if n == d.name {
+					form = flag_form.Negative
+					break
+				}
+			}
+		}
+		return nil, fmt.Errorf("option name '%s' cannot specify an option with definition '%#v'", opt, d)
+	}
+	return &optionBase{
+		Definition: d,
+		Form:       form,
+	}, nil
+}
+
+// Coerce this option base to use the standard name for this option.
+func (o *optionBase) UseName() {
+	if o.Name() == "" {
+		log.Warnf("Attempted to use name for option %s, which lacks a name.", o.Name())
+		return
+	}
+	o.Form = o.Form.AsNameType(flag_form.Standard)
+}
+
+// Coerce this option base to use the short name for this option. Does not
+// support coercing from a negative (`--no`) type flag.
+func (o *optionBase) UseShortName() {
+	if o.ShortName() == "" {
+		log.Warnf("Attempted to use short name for option %s, which lacks a short name.", o.Name())
+		return
+	}
+	if o.Form.Negative() {
+		log.Warnf("Cannot coerce short form from negative form for option %s.", o.Name())
+		return
+	}
+	o.Form = flag_form.Short
+}
+
+// Returns whether or not this option will be represented using the standard
+// name for the option.
+func (o *optionBase) UsesName() bool {
+	return o.Form.CompareNameType(flag_form.Standard)
+}
+
+// Returns whether or not this option will be represented using the short name
+// for the option.
+func (o *optionBase) UsesShortName() bool {
+	return o.ShortName() != "" && o.Form == flag_form.Short
+}
+
+// Returns a formatted version of the option.
+func (o *optionBase) Format() string {
+	switch o.Form {
+	case flag_form.Standard:
+		return "--" + o.Name()
+	case flag_form.Negative:
+		return "--" + "no" + o.Name()
+	case flag_form.Short:
+		return "-" + o.ShortName()
+	default:
+		// Just default to the standard representaation if the form is unknown.
+		return "--" + o.Name()
+	}
+}
```

Comment:
nit: it's mildly confusing to me that this function doesn't return the `=value` part.  My mental model is that `--no<name>` is just sugar for `--name=false` so it seems like we're effectively returning the flag value, but only in that one case, and not the other cases, and it seems unclear why we're doing that. This could just be a misunderstanding on my part though.

Reply:
`OptionBase` doesn't have the value, only the form and the definition, so it can't return the value. This is because it supports options, like expansion options, that do not have a value, while the negative is part of the flag's form. The handling of the formatting of the value is done by the option itself.